x-pack/plugins/security_solution/public/common/components/sourcerer/timeline_sourcerer.test.tsx

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { render, cleanup, fireEvent, screen, waitFor } from '@testing-library/react';
+import { SourcererScopeName } from '../../store/sourcerer/model';
+import { Sourcerer } from '.';
+import { sourcererModel } from '../../store/sourcerer';
+import {
+  createSecuritySolutionStorageMock,
+  kibanaObservable,
+  mockGlobalState,
+  SUB_PLUGINS_REDUCER,
+  TestProviders,
+} from '../../mock';
+import { createStore } from '../../store';
+import { useSourcererDataView } from '../../containers/sourcerer';
+import { useSignalHelpers } from '../../containers/sourcerer/use_signal_helpers';
+
+const mockDispatch = jest.fn();
+
+jest.mock('../../containers/sourcerer');
+jest.mock('../../containers/sourcerer/use_signal_helpers');
+const mockUseUpdateDataView = jest.fn().mockReturnValue(() => true);
+jest.mock('./use_update_data_view', () => ({
+  useUpdateDataView: () => mockUseUpdateDataView,
+}));
+jest.mock('react-redux', () => {
+  const original = jest.requireActual('react-redux');
+
+  return {
+    ...original,
+    useDispatch: () => mockDispatch,
+  };
+});
+
+jest.mock('@kbn/kibana-react-plugin/public', () => {
+  const original = jest.requireActual('@kbn/kibana-react-plugin/public');
+
+  return {
+    ...original,
+    toMountPoint: jest.fn(),
+  };
+});
+
+const mockUpdateUrlParam = jest.fn();
+jest.mock('../../utils/global_query_string', () => {
+  const original = jest.requireActual('../../utils/global_query_string');
+
+  return {
+    ...original,
+    useUpdateUrlParam: () => mockUpdateUrlParam,
+  };
+});
+
+const { id } = mockGlobalState.sourcerer.defaultDataView;
+
+let store: ReturnType<typeof createStore>;
+const sourcererDataView = {
+  indicesExist: true,
+  loading: false,
+};
+
+describe('timeline sourcerer', () => {
+  const { storage } = createSecuritySolutionStorageMock();
+  store = createStore(mockGlobalState, SUB_PLUGINS_REDUCER, kibanaObservable, storage);
+  const testProps = {
+    scope: sourcererModel.SourcererScopeName.timeline,
+  };
+
+  beforeEach(async () => {
+    const pollForSignalIndexMock = jest.fn();
+
+    (useSourcererDataView as jest.Mock).mockReturnValue(sourcererDataView);
+
+    (useSignalHelpers as jest.Mock).mockReturnValue({
+      pollForSignalIndex: pollForSignalIndexMock,
+      signalIndexNeedsInit: false,
+    });
+
+    render(
+      <TestProviders store={store}>
+        <Sourcerer {...testProps} />
+      </TestProviders>
+    );
+
+    fireEvent.click(screen.getByTestId('timeline-sourcerer-trigger'));
+
+    await waitFor(() => {
+      fireEvent.click(screen.getByTestId(`sourcerer-advanced-options-toggle`));
+    });
```

Comment:
yes, it was intentional. `waitFor` was used here because `sourcerer-advanced-options-toggle` takes some time to appear, otherwise this may throw the error. So I guess, it should remain as it is.